nimare/dataset/dataset.py

```diff
@@ -1,68 +1,187 @@
 # -*- coding: utf-8 -*-
 # emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
 # vi: set ft=python sts=4 ts=4 sw=4 et:
+"""
+Classes for representing datasets of images and/or coordinates.
+"""
+from __future__ import print_function
 import json
+import time
+import gzip
+import pickle
+from os import mkdir
+from os.path import join, isdir, isfile
+import pandas as pd
+from pyneurovault import api
+from ..utils import get_resource_path
+from .extract import (NeuroVaultDataSource, NeurosynthDataSource,
+                      BrainSpellDataSource)
 
-from .base import ConnMatrix, Image, ActivationSet, Surface
 
+def to_chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in xrange(0, len(l), n):
+        yield l[i:i + n]
 
-class Contrast(object):
-    """Container for contrasts (aka experiments or comparisons) nested within
-    studies.
-    Should store an arbitrary number of ConnMatrices, Images, and Surfaces,
-    along with at most one ActivationSet.
+
+def download_combined_database(out_dir, overwrite=False):
```

Comment:
Addressing an issue brought up in #2, I moved `download_combined_database` into `nimare.dataset.extract` in my fork. I can move it into something like `nimare.dataset.download`, but the related downloading functions are now in `extract`, so I figured this function belonged there too.